src/pages/itos/index.tsx

```diff
@@ -96,157 +100,63 @@ export const displayITOpacks = (
-    const itoPage = tempPage ?? page;
-    const isActive = router?.query?.active ?? true;
-    return api.get({
-      route: `ito/list`,
-      query: { page: itoPage, active: isActive, asset },
-    });
-  };
-
-  const requestITOsAndReset = async () => {
-    const res = await requestITOs(1);
-    if (!res.error || res.error === '') {
-      const newITOs = res?.data?.itos || [];
-      await parseITOs(newITOs);
-      setPage(1);
-      setLastPage(res.pagination.totalPages);
-      setITOs(newITOs);
-    }
-  };
-
-  const requestITOsAndAddMore = async () => {
-    const res = await requestITOs();
-    if (!res.error || res.error === '') {
-      const newITOs = res?.data?.itos || [];
-      await parseITOs(newITOs);
-      const allITOs = [...ITOs, ...newITOs];
-      setLastPage(res.pagination.totalPages);
-      setITOs(allITOs);
-    }
-  };
-
-  const parseITOs = async (ITOs: IITO[]): Promise<IParsedITO | never[]> => {
-    const assetsInput: string = ITOs.map(ITO => ITO.assetId).join(',');
-    const packsPrecisionCalls: Promise<IITO>[] = [];
-    const res = await api.get({
-      route: `assets/list?asset=${assetsInput}`,
-    });
-    if (!res.error || res.error === '') {
-      const assets = res.data.assets;
-      ITOs.forEach((ITO, index) => {
-        const asset = assets.find(
-          (asset: IAsset) => asset.assetId === ITOs[index].assetId,
-        );
-        ITO.maxAmount = ITO.maxAmount / 10 ** asset.precision;
-        ITO['ticker'] = asset.ticker;
-        ITO['assetType'] = asset.assetType;
-        ITO['precision'] = asset.precision;
-        ITO['assetLogo'] = asset.logo;
-        packsPrecisionCalls.push(processITOPrecisions(ITO, asset.precision));
-      });
-      await Promise.allSettled(packsPrecisionCalls);
-    }
-    return [];
-  };
+  const {
+    data,
+    fetchNextPage: nextPageITOs,
+    isFetching,
+    hasNextPage,
+    isLoading: isLoadingITOs,
+    refetch: refetchITOs,
+  } = useInfiniteQuery(
+    'ITOss',
+    async ({ pageParam = 1 }) => {
+      const dataITOs = await requestITOs(router, pageParam);
+      const assets = await requestAssetsList(dataITOs);
+      await parseITOsRequest(dataITOs, assets);
+      return dataITOs;
+    },
+    {
+      getNextPageParam: lastPage => {
+        if (lastPage) {
+          const { self, totalPages } = lastPage?.pagination;
+          if (totalPages > self) {
+            return self + 1;
+          }
+        }
+      },
+      enabled: !!router.isReady,
+    },
+  );
 
-  const processITOPrecisions = async (
-    ITO: IITO,
-    assetPrecision: number,
-  ): Promise<IITO> => {
-    ITO.packData.forEach(async packInfo => {
-      const keyPrecision = await getPrecision(packInfo.key);
-      packInfo.packs.forEach(pack => {
-        pack.price = pack.price / 10 ** keyPrecision;
-        pack.amount = pack.amount / 10 ** assetPrecision;
-      });
-    });
-    return ITO;
-  };
+  const itemsITOs = useMemo(() => {
```

Comment:
dont use useMemo for iterations below 1k items. Consider reading [here](https://react.dev/learn/you-might-not-need-an-effect#how-to-tell-if-a-calculation-is-expensive)